go/interpreter/lfvm/instructions.go

```diff
@@ -989,6 +979,19 @@ func genericCreate(c *context, kind tosca.CallKind) {
 	c.refund += res.GasRefund
 }
 
+// checkInitCodeSize checks the size of the init code.
+// An error is returned if size is greater than MaxInitCodeSize.
+func checkMaxInitCodeSize(size uint64) error {
```

Comment:
This is ok, but it would be much better if this function offloads some work from the caller side: 

i.e.  `computeCodeSizeCost(...) (gas,error)`
Where the function is responsible for all the semantics related to code size: revision check, cost computations, overflows during computation.

This would streamline the op code and centralize all the side-effects related to init code .